aws/resource_aws_dynamodb_table_test.go

```diff
@@ -14,6 +16,67 @@ import (
 	"github.com/hashicorp/terraform/terraform"
 )
 
+func init() {
+	resource.AddTestSweepers("aws_dynamodb_table", &resource.Sweeper{
+		Name: "aws_dynamodb_table",
+		F:    testSweepDynamoDbTables,
+	})
+}
+
+func testSweepDynamoDbTables(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).dynamodbconn
+
+	prefixes := []string{
+		"TerraformTest",
+		"tf-acc-test-",
+		"tf-autoscaled-table-",
+	}
+
+	err = conn.ListTablesPages(&dynamodb.ListTablesInput{}, func(out *dynamodb.ListTablesOutput, lastPage bool) bool {
+		for _, tableName := range out.TableNames {
+			hasPrefix := false
+			for _, prefix := range prefixes {
+				if strings.HasPrefix(*tableName, prefix) {
+					hasPrefix = true
+				}
+			}
+			if !hasPrefix {
+				continue
+			}
```

Comment:
🤔  Can't we just simplify the whole block of code above to something like this?

```go
for _, prefix := range prefixes {
	if !strings.HasPrefix(*tableName, prefix) {
		continue
	}
}
```

Reply:
Ha yes it can. Copypasta 🍝

Reply:
Pushed up this refactor and added a log line there too 👍